eng/pipelines/templates/stages/archetype-android-release.yml

```diff
@@ -0,0 +1,125 @@
+parameters:
+  Artifacts: []
+  ArtifactName: 'not-specified'
+
+stages:
+  # The signing stage is responsible for submitting binaries to ESRP for our official signing
+  # where appropriate and also meeting any other signing requirements for particular artifacts,
+  # in this case we do GPG signing in order to publish to Maven Central. At the moment signing
+  # is protected by an approval check but this may be removed in the future.
+  - stage: Signing
+    dependsOn: ${{parameters.DependsOn}}
+    jobs:
+      - deployment: SignPackage
+        environment: esrp
+        timeoutInMinutes: 20
+        pool:
+          vmImage: ubuntu-16.04
+
+        strategy:
+          runOnce:
+            deploy:
+              steps:
+                - checkout: azure-sdk-build-tools
+                  path: azure-sdk-build-tools
+
+                - download: current
+                  artifact: ${{parameters.ArtifactName}}
+                  timeoutInMinutes: 5
+
+                - template: /tools/java-esrp-signing/java-esrp-signing.yml@azure-sdk-build-tools
+                  parameters:
+                    Artifacts: ${{parameters.Artifacts}}
+                    ArtifactDirectory: $(Pipeline.Workspace)/${{parameters.ArtifactName}}
+
+                - publish: $(Pipeline.Workspace)/${{parameters.ArtifactName}}
+                  artifact: ${{parameters.ArtifactName}}-signed
+                  displayName: 'Store signed packages in ${{parameters.ArtifactName}}-signed artifact'
+                  timeoutInMinutes: 5
+
+  # We generate two interdepdent stages for each artifact listed in the ci.yml file, creates the release
+  # in GitHub. The Release stage publishes to Maven Central. Both stages require approval since they
+  # effectively burn the version number. For testing of packages prior to burning the version number -
+  # the Validation step below publishes a package to a "burner" feed which is cleaned up after the
+  # pipeline completes.
+  - ${{if and(eq(variables['Build.Reason'], 'Manual'), eq(variables['System.TeamProject'], 'internal'))}}:
+    - ${{ each artifact in parameters.Artifacts }}:
+      - stage: Release_${{artifact.safeName}}
+        displayName: 'Release: ${{artifact.name}}'
+        dependsOn: Signing
+        condition: and(succeeded(), ne(variables['SetDevVersion'], 'true'), ne(variables['Skip.Release'], 'true'), ne(variables['Build.Repository.Name'], 'Azure/azure-sdk-for-android-pr'))
+        jobs:
+          - deployment: TagRepository
+            displayName: "Create release tag"
+            condition: ne(variables['Skip.TagRepository'], 'true')
+            environment: github
+            timeoutInMinutes: 5
+            dependsOn:
+              - ${{ if eq(parameters.VerifyVersions, 'true')}}:
+                - VerifyReleaseVersion
+
+            pool:
+              vmImage: vs2017-win2016
```

Comment:
Looks like we need to probably update most of the images

Reply:
Yeah it is all a bit stale. My goal here though is to get this stuff moved over first and then start doing a bit of a facelift.